xgcm/grid.py

```diff
@@ -1121,6 +1136,7 @@ def diff(self, da, axis, **kwargs):
         """
 
         ax = self.axes[axis]
+        kwargs.setdefault("boundary", self.boundary[axis])
```

Comment:
I think this should be consistently applied (also for `.min()` and `.max()`. Is there a clever way to generalize all these functions so we avoid the repeated code?

Reply:
I guess a decorator could be used. I've implemented manually for now.

Reply:
I am very unfamiliar with decorators but I am very interested. Perhaps we can address this together with #157.